cmake/pika_setup_boost.cmake

```diff
@@ -46,7 +46,8 @@ if(NOT TARGET pika_dependencies_boost)
 
   list(REMOVE_DUPLICATES __boost_libraries)
 
-  find_package(Boost ${Boost_MINIMUM_VERSION} MODULE REQUIRED COMPONENTS ${__boost_libraries})
+  pika_set_cmake_policy(CMP0167 NEW)  # deprecates FindBoost
+  find_package(Boost ${Boost_MINIMUM_VERSION} CONFIG REQUIRED COMPONENTS ${__boost_libraries})
```

Comment:
I actually see we still have a `find_package(Boost)` without an explicit `MODULE` or `CONFIG` above this on line 35. Should we be explicit about `CONFIG` there as well, or remove `CONFIG` from both?